index.js

```diff
@@ -204,59 +204,73 @@ function isSourceElement(opts, node) {
   if (node == null) {
     return false;
   }
-  switch (node.type || node.kind) {
-    case "ObjectExpression": // JSON
-    case "ArrayExpression": // JSON
-    case "StringLiteral": // JSON
-    case "NumericLiteral": // JSON
-    case "BooleanLiteral": // JSON
-    case "NullLiteral": // JSON
-    case "FunctionDeclaration":
-    case "BlockStatement":
-    case "BreakStatement":
-    case "ContinueStatement":
-    case "DebuggerStatement":
-    case "DoWhileStatement":
-    case "EmptyStatement":
-    case "ExpressionStatement":
-    case "ForInStatement":
-    case "ForStatement":
-    case "IfStatement":
-    case "LabeledStatement":
-    case "ReturnStatement":
-    case "SwitchStatement":
-    case "ThrowStatement":
-    case "TryStatement":
-    case "VariableDeclaration":
-    case "WhileStatement":
-    case "WithStatement":
-    case "ClassDeclaration": // ES 2015
-    case "ImportDeclaration": // Module
-    case "ExportDefaultDeclaration": // Module
-    case "ExportNamedDeclaration": // Module
-    case "ExportAllDeclaration": // Module
-    case "TypeAlias": // Flow
-    case "InterfaceDeclaration": // Flow, Typescript
-    case "TypeAliasDeclaration": // Typescript
-    case "ExportAssignment": // Typescript
-    case "ExportDeclaration": // Typescript
-    case "OperationDefinition": // GraphQL
-    case "FragmentDefinition": // GraphQL
-    case "VariableDefinition": // GraphQL
-    case "TypeExtensionDefinition": // GraphQL
-    case "ObjectTypeDefinition": // GraphQL
-    case "FieldDefinition": // GraphQL
-    case "DirectiveDefinition": // GraphQL
-    case "EnumTypeDefinition": // GraphQL
-    case "EnumValueDefinition": // GraphQL
-    case "InputValueDefinition": // GraphQL
-    case "InputObjectTypeDefinition": // GraphQL
-    case "SchemaDefinition": // GraphQL
-    case "OperationTypeDefinition": // GraphQL
-    case "InterfaceTypeDefinition": // GraphQL
-    case "UnionTypeDefinition": // GraphQL
-    case "ScalarTypeDefinition": // GraphQL
-      return true;
+  // JS and JS like to avoid repetitions
+  const jsSourceElements = [
```

Comment:
What's the rule with global variables? those `const ...SourceElements` could be defined once.